chatsky/pipeline/service/group.py

```diff
@@ -61,26 +52,41 @@ class ServiceGroup(PipelineComponent, extra="forbid", arbitrary_types_allowed=Tr
             ServiceGroup,
         ]
     ]
+    """
+    A `ServiceGroup` object, that will be added to the group.
+    """
+    # Inherited fields repeated. Don't delete these, they're needed for documentation!
+    before_handler: BeforeHandler = Field(default_factory=BeforeHandler)
+    after_handler: AfterHandler = Field(default_factory=AfterHandler)
+    timeout: Optional[float] = None
+    requested_async_flag: Optional[bool] = None
+    start_condition: StartConditionCheckerFunction = Field(default=always_start_condition)
+    name: Optional[str] = None
+    path: Optional[str] = None
 
     @model_validator(mode="before")
     @classmethod
-    # Here Script class has "@validate_call". Is it needed here?
-    def components_constructor(cls, data: Any):
-        if not isinstance(data, dict):
+    def __components_constructor(cls, data: Any):
+        if isinstance(data, (list, PipelineComponent, Callable)):
             result = {"components": data}
-        else:
+        elif isinstance(data, dict):
             result = data.copy()
+        else:
+            raise ValueError(
+                "Service Group can only be initialized from a Dict,"
+                " a PipelineComponent or a list of PipelineComponents. Wrong inputs received."
+            )
 
         if ("components" in result) and (not isinstance(result["components"], list)):
             result["components"] = [result["components"]]
         return result
 
     @model_validator(mode="after")
-    def calculate_async_flag(self):
+    def __calculate_async_flag(self):
         self.calculated_async_flag = all([service.asynchronous for service in self.components])
         return self
 
-    async def run_component(self, ctx: Context, pipeline: Pipeline) -> None:
+    async def run_component(self, ctx: Context, pipeline: Pipeline) -> Optional[ComponentExecutionState]:
         """
         Method for running this service group. Catches runtime exceptions and logs them.
```

Comment:
Do you need to copy the docstring?
I think the docstring should be inherited from `PipelineComponent` in the docs.

Reply:
True, I see ComponentExecutionState being described here, which is the only difference from PipelineComponent's docstring, but it's fairly obvious, could remove this.

Reply:
Yeah, ok. If the component docstring is going to be as short as I suggested in the other comment then every concrete component should describe what it is doing during `run_component`.